src/vehicle.cpp

```diff
@@ -5806,7 +5746,7 @@ bool vehicle::explode_fuel( int p, damage_type type )
             name.c_str());
         g->explosion( global_part_pos3( p ), pow, 0.7, data.fiery_explosion );
         parts[p].hp = 0;
-        parts[p].amount = 0;
+        parts[p].ammo_set( "null", 0 );
```

Comment:
No good reason and considering I wrote both implementations no good excuse either.